worlds/checksmate/Options.py

```diff
@@ -0,0 +1,415 @@
+    replace the Bishop and Knight with a Vao and Cannon, constitute a very powerful yet flawed Different Army.
+
+    Chaos: Chooses random enabled options.
+
+    Limited: Chooses within your army. (If you want at most 2 Bishops, 2 Knights, 2 Rooks, and 1 Queen, add Piece Type
+    Limits below: 2 Minor, 2 Major, and 1 Queen.)
+    """
+    display_name = "Fairy Chess Army"
+    option_chaos = 0
+    option_limited = 1
+    default = 0
+
+
+class FairyChessPawns(Choice):
+    """
+    Whether to use fairy chess pawns.
+
+    Vanilla: Only use the standard pawn.
+
+    Mixed: Adds all implemented fairy chess pawns to the pool. You may receive a mix of different types of pawns.
+
+    Berolina: Only use the Berolina pawn (may appear to be a Ferz), which moves diagonally and captures forward.
+    """
+    display_name = "Fairy Chess Pawns"
+    option_vanilla = 0
+    option_mixed = 1
+    option_berolina = 2
+    default = 1
+
+
+class FairyKings(Range):
+    """
+    Whether to use fairy king upgrades, such as the Knight's moves.
+    """
+    display_name = "Fairy Kings"
+    range_start = 0
+    range_end = 2
+    default = 0
+
+
+class MinorPieceLimitByType(SpecialRange):
+    """
+    How many of any given type of minor piece you might play with. If set to 1, you will never start with more than 1
+    Knight, nor 1 Bishop, but you may have both 1 Knight and 1 Bishop. If set to 0, this setting is disabled.
+    """
+    display_name = "Minor Piece Limit by Type"
+    range_start = 0
+    range_end = 9
+    default = 0
+    special_range_cutoff = 1
+
+
+class MajorPieceLimitByType(SpecialRange):
+    """
+    How many of any given type of major piece you might play with. If set to 1, you will never start with more than 1
+    Rook. If set to 0, this setting is disabled.
+    """
+    display_name = "Major Piece Limit by Type"
+    range_start = 0
+    range_end = 9
+    default = 0
+    special_range_cutoff = 1
+
+
+class QueenPieceLimitByType(SpecialRange):
+    """
+    How many of any given type of Queen-equivalent piece you might play with. If set to 1, you will never start with
+    more than 1 Queen. You may have both 1 Queen and 1 Amazon. If set to 0, this setting is disabled.
+    """
+    display_name = "Queen Piece Limit by Type"
+    range_start = 0
+    range_end = 5
+    default = 0
+    special_range_cutoff = 1
+
+
+class PocketLimitByPocket(SpecialRange):
+    """
+    How many Progressive Pocket items might be allocated to any given pocket. If this is set to 1, any given Pocket will
+    never hold anything more substantial than a Pawn. If this is set to 3, any given Pocket will never hold a Queen.
+
+    The default of 4 allows each of the 3 spaces to hold between 0-4 progressive items.
+    """
+    display_name = "Pocket Limit by Pocket"
+    range_start = 0
+    range_end = 4
+    default = 4
+    special_range_cutoff = 1
+
+
+class QueenPieceLimit(SpecialRange):
```

Comment:
As of current main, Special Range is deprecated. You'll need to move to NamedRange.
To get this behavior, you'll need to either override from_text to catch the "random" case, or set range_start to 1 and define a name at 0 to define it as the disabled option.

Reply:
Thanks for the example, it was extremely helpful in porting to a non-deprecated form.